python/chronos/src/bigdl/chronos/data/utils/export_torchscript.py

```diff
@@ -0,0 +1,184 @@
+
+
+class ExportJIT(nn.Module):
+    def __init__(self, lookback: int, id_index: int,
+                 target_feature_index: List[int]) -> None:
+        super().__init__()
+        self.lookback = lookback
+        self.target_feature_index = target_feature_index
+        self.id_index = id_index
+
+    def _shift(self, data, i: int):
+        res = torch.empty_like(data, dtype=torch.float32)
+        if i < 0:
+            res[:] = torch.roll(data, i, 0)
+            res[i:] = torch.nan
+        elif i == 0:
+            res[:] = data
+        return res
+
+    def _groupby(self, data, colunm):
+        # group the data using unique values in column
+        # This code is adapted from https://github.com/pytorch/pytorch/issues/20613
+
+        sorted_values, order = colunm.sort(0)
+        delta = colunm[1:] - colunm[:-1]
+        non_zero = delta.nonzero()
+
+        if len(non_zero) == 0:
+            return [data]
+
+        cutpoints = non_zero[0]
+        cutpoints: List[int] = cutpoints.add(1).tolist()
+        res: List[torch.Tensor] = []
+        for start, end in zip([0] + cutpoints, cutpoints + [len(colunm)]):
+            index, _ = order[start:end].sort(0)
+            res.append(data[index])
+
+        return res
+
+    def _roll_tensor(self, data, lookback: int, target_feature_index: List[int]):
+        data = data[:, target_feature_index]
+        data = torch.unsqueeze(data, 1)
+        roll_data = torch.cat([self._shift(data, i) for i in range(0, -lookback, -1)],
+                              dim=1)
+        window_idx = torch.arange(lookback)
+        if data.size()[0] >= lookback:
+            roll_data = roll_data[:data.size()[0]-lookback+1, window_idx, :]
+        else:
+            roll_data = roll_data[:0, window_idx, :]
+        return roll_data
+
+    def roll(self, data, lookback: int, id_index: int, target_feature_index: List[int]):
+        id_col = data[:, id_index]
+        res: List[torch.Tensor] = self._groupby(data, id_col)
+        roll_result: List[torch.Tensor] = []
+        for group in res:
+            roll_result.append(self._roll_tensor(group, lookback, target_feature_index))
+
+        return torch.cat(roll_result, dim=0)
+
+    def scale(self, data):
+        return data
+
+    def export_preprocessing(self, data):
+        data[:, self.target_feature_index] = self.scale(data[:, self.target_feature_index])
+        data_roll = self.roll(data, self.lookback, self.id_index, self.target_feature_index)
+        return data_roll
+
+    def forward(self, data):
+        return self.export_preprocessing(data)
+
+
+class ExportWithStandardScaler(ExportJIT):
+    def __init__(self, scaler: StandardScaler, lookback: int,
+                 id_index: int, target_feature_index: List[int]) -> None:
+        super().__init__(lookback, id_index, target_feature_index)
+        self.scale_ = torch.from_numpy(scaler.scale_).type(torch.float64)
+        self.mean_ = torch.from_numpy(scaler.mean_).type(torch.float64)
```

Comment:
why are we using float64 here, I may have forgotten part of our discussion :(

Reply:
It's mainly used to make the results consistent with the output when using original Chronos API. 

The type of `self.scale_` is `List[float]` at the beginning, but it can't handle the case when the input type is float64, so I changed the type of `self.scale` to a float64 tensor.